s3po/backends/s3.py

```diff
@@ -0,0 +1,94 @@
+'''Deal with S3.'''
+
+
+from boto.s3.connection import S3Connection
+from cStringIO import StringIO
+
+from ..util import CountFile, retry, logger
+from ..exceptions import UploadException, DownloadException
+
+
+class S3(object):
+    '''Our connection to S3'''
+    # How big must a file get before it's multiparted. Also how big the chunks
+    # are that we'll read
+    chunk_size = 50 * 1024 * 1204
+    min_chunk_size = 5 * 1024 * 1024
+
+    def __init__(self, *args, **kwargs):
+        self.conn = S3Connection(*args, **kwargs)
+
+    def get_bucket(self, bucket):
+        return self.conn.get_bucket(bucket, validate=False)
+
+    def download(self, bucket, key, fobj, retries):
```

Comment:
We need to be able to pass headers to the download as well. For example, if we want to pass a `Range` header.